File: leetcode/js/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for(let i = 0; i < nums.length - 1; i++) {
        let j = i + 1;
        for(j; j< nums.length; j++) {
            if(nums[i] + nums[j] == target) return [i, j];
        }
    }
};

File: leetcode/go/1-two-sum.go
package leetcode

func twoSum(nums []int, target int) []int {
	m := make(map[int]int, len(nums)-1)
	for i, num := range nums {
		if v, ok := m[target-num]; ok {
			return []int{v, i}
		}
		m[num] = i
	}
	return nil
}
